content/en/docs/instrumentation/cpp/getting-started.md

```diff
@@ -1,185 +1,393 @@
+  }
+};
+
+void run() {
+  auto router = oatpp::web::server::HttpRouter::createShared();
+  router->route("GET", "/rolldice", std::make_shared<Handler>());
+  auto connectionHandler = oatpp::web::server::HttpConnectionHandler::createShared(router);
+  auto connectionProvider = oatpp::network::tcp::server::ConnectionProvider::createShared({"localhost", 8080, oatpp::network::Address::IP_4});
+  oatpp::network::Server server(connectionProvider, connectionHandler);
+  OATPP_LOGI("Dice Server", "Server running on port %s", connectionProvider->getProperty("port").getData());
+  server.run();
+}
+
+int main() {
+  oatpp::base::Environment::init();
+  run();
+  oatpp::base::Environment::destroy();
+  return 0;
+}
 
-```sh
-sudo yum install git cmake g++ libcurl-devel
 ```
 
-{{% /tab %}} {{% tab "Linux (alpine)" %}}
+Build and run the application with the following CMake commands.
 
-```sh
-sudo apk add git cmake g++ make curl-dev
+```bash
+mkdir build
+cd build 
+cmake ..
+cmake --build .
 ```
 
-{{% /tab %}} {{% tab "MacOS (homebrew)" %}}
+After successfully building your project, you can run the generated executable.
 
-```sh
-xcode-select —install
-/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
-brew install git cmake
+```bash
+./dice-server
 ```
 
-{{% /tab %}} {{< /tabpane >}}
+Then, open <http://localhost:8080/rolldice> in your browser to ensure it is working.
 
-## Building
 
-Get the `opentelemetry-cpp` source:
+## Instrumentation 
 
-```shell
-git clone --recursive https://github.com/open-telemetry/opentelemetry-cpp
-```
 
-Navigate to the repository cloned above, and create the CMake build
-configuration:
+To add OpenTelemetry to your application, update the `CMakeLists.txt` file with the
+following additional dependencies. 
 
-```shell
-cd opentelemetry-cpp
-mkdir build && cd build
-cmake -DBUILD_TESTING=OFF -DWITH_EXAMPLES_HTTP=ON ..
-```
+```cmake
+project(RollDiceServer)
+cmake_minimum_required(VERSION 3.1)
+# Set C++ standard (e.g., C++17)
+set(CMAKE_CXX_STANDARD 17)
+set(project_name roll-dice-server)
 
-Once build configuration is created, build the CMake targets `http_client` and
-`http_server`:
+# Define your project's source files
+set(SOURCES
+    main.cpp  # Add your source files here
+)
+# Create an executable target
+add_executable(dice-server ${SOURCES})
 
-```shell
-cmake --build . --target http_client http_server
-```
+set(OATPP_ROOT ../oatpp)
+set(OPENTELEMETRY_ROOT ../opentelemetry-cpp)
+find_library(OATPP_LIB NAMES liboatpp.a HINTS "${OATPP_ROOT}/build/src/" NO_DEFAULT_PATH)
+if (NOT OATPP_LIB)
+	message(SEND_ERROR "Did not find oatpp library ${OATPP_ROOT}/build/src")
+endif()
+# set the path to the directory containing "oatpp" package configuration files
+include_directories(${OATPP_ROOT}/src)
 
-If all goes well, you should find binaries `http_server` and `http_client` in
-`./examples/http`:
+include_directories(${OPENTELEMETRY_ROOT}/api/include)
```

Comment:
Could here use `find_package(opentelemetry-cpp CONFIG)` instead of adding include directories 1 by 1?

Reply:
I'm unfamiliar with `find_package`. I tried importing targets with find_package first but ran into issues. `find_package(opentelemetry-cpp CONFIG)` this would replace lines 251 through 254? anything else need to change as well?

Reply:
@ThomsonTan can you provide some guidance what needs to change for this? Or can we fix this in a follow up PR?

Reply:
@Akhaled19 We can fix this via a follow up issue.

Reply:
> @ThomsonTan I'm unfamiliar with `find_package`. I tried importing targets with find_package first but ran into issues. `find_package(opentelemetry-cpp CONFIG)` this would replace lines 251 through 254? anything else need to change as well?

That's correct. Can you please share what issue you get for switching to `find_pacakge(...)`?

Fix this in follow up issue is also fine.

Reply:
I'm sure with some fixes I can resolve the issue but it would take me some time. Here is the error I get when I switch these lines to `find_package(opentelemetry-cpp CONFIG)` : 
```
Could not find a package configuration file provided by "opentelemetry-cpp"
  with any of the following names:

    opentelemetry-cppConfig.cmake
    opentelemetry-cpp-config.cmake
```

Reply:
I'll create an issue for that